core/drivers/bcm_gpio.c

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2019 Broadcom.
+ */
+#include <assert.h>
+#include <gpio.h>
+#include <initcall.h>
+#include <io.h>
+#include <kernel/misc.h>
+#include <mm/core_memprot.h>
+#include <mm/core_mmu.h>
+#include <platform_config.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <trace.h>
+
+#define BCM_IPROC_GPIO  "bcm-iproc-gpio"
+
+#define IPROC_GPIO_DATA_IN_OFFSET   0x00
+#define IPROC_GPIO_DATA_OUT_OFFSET  0x04
+#define IPROC_GPIO_OUT_EN_OFFSET    0x08
+#define IPROC_GPIO_INT_MSK_OFFSET   0x18
+
+#define GPIO_BANK_SIZE  0x200
+#define NGPIOS_PER_BANK 32
+#define GPIO_BANK(pin)  ((pin) / NGPIOS_PER_BANK)
+
+#define IPROC_GPIO_REG(pin, reg) (GPIO_BANK(pin) * GPIO_BANK_SIZE + (reg))
+#define IPROC_GPIO_SHIFT(pin)    ((pin) % NGPIOS_PER_BANK)
+
+static inline void iproc_set_bit(struct gpio_chip *gc, unsigned int reg,
```

Comment:
do not use `inline` for static function. let the compiler choose the best strategy according to build directives.

Reply:
done.